server/lib/flow-event.js

```diff
@@ -32,26 +124,17 @@ module.exports = function (event, data, request) {
     hostname: HOSTNAME,
     op: 'flowEvent',
     pid: process.pid,
-    time: event.time,
+    time: new Date(event.time).toISOString(),
     userAgent: request.headers['user-agent'],
     v: VERSION
   }, _.mapValues(pickedData, sanitiseData));
 
   optionallySetFallbackData(eventData, 'service', data.client_id);
   optionallySetFallbackData(eventData, 'entrypoint', data.entryPoint);
 
-  if (typeof eventData.time === 'number') {
```

Comment:
When validation landed, it became impossible for this not to be a number, hence removing the condition and marshalling it directly in the object literal.